src/java.desktop/share/classes/sun/java2d/pipe/OutlineTextRenderer.java

```diff
@@ -67,47 +66,26 @@ public void drawChars(SunGraphics2D g2d,
                           char[] data, int offset, int length,
                           int x, int y) {
 
-        String s = new String(data, offset, length);
-        drawString(g2d, s, x, y);
+        StandardGlyphVector sgv = new StandardGlyphVector(g2d.getFont(), data, offset, length, g2d.getFontRenderContext());
+        drawGlyphVector(g2d, sgv, (float) x, (float) y);
     }
 
     public void drawString(SunGraphics2D g2d, String str, double x, double y) {
 
-        if ("".equals(str)) {
-            return; // TextLayout constructor throws IAE on "".
-        }
-        TextLayout tl = new TextLayout(str, g2d.getFont(),
-                                       g2d.getFontRenderContext());
-        Shape s = tl.getOutline(AffineTransform.getTranslateInstance(x, y));
-
-        int textAAHint = g2d.getFontInfo().aaHint;
-
-        int prevaaHint = - 1;
-        if (textAAHint != SunHints.INTVAL_TEXT_ANTIALIAS_OFF &&
-            g2d.antialiasHint != SunHints.INTVAL_ANTIALIAS_ON) {
-            prevaaHint = g2d.antialiasHint;
-            g2d.antialiasHint =  SunHints.INTVAL_ANTIALIAS_ON;
-            g2d.validatePipe();
-        } else if (textAAHint == SunHints.INTVAL_TEXT_ANTIALIAS_OFF
-            && g2d.antialiasHint != SunHints.INTVAL_ANTIALIAS_OFF) {
-            prevaaHint = g2d.antialiasHint;
-            g2d.antialiasHint =  SunHints.INTVAL_ANTIALIAS_OFF;
-            g2d.validatePipe();
-        }
-
-        g2d.fill(s);
-
-        if (prevaaHint != -1) {
-             g2d.antialiasHint = prevaaHint;
-             g2d.validatePipe();
-        }
+        StandardGlyphVector sgv = new StandardGlyphVector(g2d.getFont(), str, g2d.getFontRenderContext());
```

Comment:
I started using `GlyphVector` instead of `TextLayout` because it gives more consistent look for animated text. For example, earlier text which has ligatures rendered with `drawString` was drawn with ligatures on `OutlineTextRenderer` and without on `GlyphListPipe`, which caused artifacts when text size crosses 100 threshold.

Reply:
Oh I see this is incorrect, I will try to come up with a different approach